src/measure_theory/integral/lebesgue.lean

```diff
@@ -2214,6 +2214,12 @@ lemma measure_preserving.set_lintegral_comp_emb {mb : measurable_space β} {ν :
 by rw [← hg.set_lintegral_comp_preimage_emb hge, preimage_image_eq _ hge.injective]
 
 section dirac_and_count
+
+@[priority 100]
+instance _root_.measurable_space.top.measurable_singleton_class {α : Type*} :
```

Comment:
I'm not sure whether this instance is dangerous. The linter wanted to a lower priority for it.

The instance itself is `Prop`-valued, so surely always the correct one, but I don't know if it affects the performance of type class inference. I'd hope not, because only the type on which the `top : measurable_space _` lives may have to be inferred, but the instance applies to nothing else but the `top` measurable space (i.e., the power set sigma algebra).

The main reason I'd want this instance is that one can transport all measure theory results (say convergence theorems, Fubini's theorem, and in this PR Markov's inequality) to `tsum` using counting measure on this obvious measurable space. And when using `tsum`, one typically doesn't want to carry around any `measurable_space` instance on the type.

Reply:
This instance won't cause loops, but I could imagine that it causes a (small) performance issue, since for every `measurable_singleton_class a inst` problem, it will try to unify `inst` with `⊤`. Type-class inference is constantly unifying instances, but typically these unifications will always succeed (at least they should since we try to avoid diamonds). Maybe make the priority even lower (`10`), and if mathlib compiles it should be fine.